gtsam/base/tests/testStdOptionalSerialization.cpp

```diff
@@ -60,8 +60,6 @@ class TestOptionalStruct {
   TestOptionalStruct() = default;
   TestOptionalStruct(const int& opt)
       : opt(opt) {}
-  // A copy constructor is needed for serialization
-  TestOptionalStruct(const TestOptionalStruct& other) = default;
```

Comment:
I think I just reviewed a PR that *added* these?

Reply:
This copy constructor was added 2 years ago.

I am currently getting an error because of this
```
error: definition of implicit copy assignment operator for 'TestOptionalStruct' is deprecated because it has a user-declared copy constructor [-Werror,-Wdeprecated-copy]
``	`
Seems to be that using the compiler generated copy-constructors and using the `= default` constructor no longer gel well together.